main.cpp

```diff
@@ -252,76 +325,128 @@ struct Dog
         int checkDogForTicks(bool extraThorough = false);
     };
 
-    // roll over
     void rollOver(bool toTheLeft);
-    // greet owners
     void greetOwner(Owner owner);
-    // go to sleep
     void goToSleep(bool pleasantDreams);
 
-    Owner owner;
+    Owner ownerNancy;
 };
 
-/*
-Thing 4) Steam Locomotive
-5 properties:
-    1) number of cars on train
-    2) amount of coal used per day
-    3) age
-    4) horsepower
-    5) number of crew on locomotive
-3 things it can do:
-    1) detach from currently coupled train car
-    2) add more coal to the furnace
-    3) slam on the brakes
- */
+void Dog::rollOver(bool toTheLeft)
+{  
+    std::string rollDirection;
+    if(toTheLeft)
+    {
+        rollDirection = "left.";
+    }
+    else
+    {
+        rollDirection = "right.";
+    }
+
+    std::cout << "Rolled over to the " << rollDirection << std::endl;   
+}
+
+void Dog::greetOwner(Owner owner)
+{
+    std::cout << "Said hello to " << owner.name << std::endl; 
+}
 
+void Dog::goToSleep(bool pleasantDreams)
+{
+    std::cout << "Went to sleep..." << std::endl;
+
+    if(pleasantDreams)
+    {
+        std::cout << "...and slept peacefully." << std::endl;
+    }
+    else
+    {
+        std::cout << "...but had nightmares." << std::endl;
+    }
+}
+
+
+void Dog::Owner::walkTheDog(float distanceToTravel, float initialSpeed)
+{
+    std::cout << "Took the dog for a walk, starting at " << initialSpeed << "km/hr and traveling " << distanceToTravel << "km." << std::endl;
+}
+
+void Dog::Owner::feedTheDog(float amountOfFood, bool isDryFood)
+{
+    std::string foodType;
+    if(isDryFood)
+    {
+        foodType = "dry food";
+    }
+    else
+    {
+        foodType = "wet food";
+    }
+
+    std::cout << "Fed the dog " << amountOfFood << "lbs of " << foodType << std::endl;
+}
+
+int Dog::Owner::checkDogForTicks(bool extraThorough)
+{
+    if(extraThorough)
+    {
+        return 3;
+    }
+    else
+    {
+        return 2;
+    }
```

Comment:
If you find yourself writing code like this:
```
int Type::someFunction()
{
    if( someExpression )
    {
        someStatement();
        return aValue;
    } 
    else 
    {
        return anotherValue;
    }
}
```
When the `if()` block is entered, the `return aValue;` is the last line of code to be executed.
you have no code after the `else` block that needs executing, since your `else` block contains a `return` statement.
`return` statements are the last thing that **ever happens** in a function
That means if your `if()` block is **NOT** entered, it will be skipped.  What is the next thing that happens?  `return anotherValue;`
that means you do **NOT** need the else block, and it is just causing visual clutter.
```
int Type::someFunction()
{
    if( someExpression )
    {
        someStatement();
        return aValue;
    } 
    return anotherValue;
}
```
- [ ] revise